lib/loader/loader.js

```diff
@@ -217,6 +217,11 @@ var loader, define, requireModule, require, requirejs;
   };
 
   define = function(name, deps, callback) {
+    var module = registry[name];
+    if (module && module.state !== 'new' && module.state !== 'errored') {
```

Comment:
Sure. My original thinking was "Since the module body never executes, then we should be able to redefine it", but I am not super adamant about this position. I'm open to whatever, just as long as it is specified.